internal/webhook/kubernetes/promotiontask/webhook.go

```diff
@@ -34,18 +34,50 @@ func SetupWebhookWithManager(
 		Complete()
 }
 
-func (w *webhook) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
-	template := obj.(*kargoapi.PromotionTask) // nolint: forcetypeassert
-	if err := libWebhook.ValidateProject(ctx, w.client, promotionTaskGroupKind, template); err != nil {
+func (w *webhook) ValidateCreate(
+	ctx context.Context,
+	obj runtime.Object,
+) (admission.Warnings, error) {
+	task := obj.(*kargoapi.PromotionTask) // nolint: forcetypeassert
+	if err := libWebhook.ValidateProject(
+		ctx,
+		w.client,
+		promotionTaskGroupKind,
+		task,
+	); err != nil {
 		return nil, err
 	}
-	return nil, nil
+	return w.validateCreateOrUpdate(task)
```

Comment:
Should these be joined together? That allows the user to see as many issues on first apply as possible. (The same applies to the other validators).

Reply:
The thing is that the error returned from ValidateProject() is never a field.Error. So it's very hard to combine it with field.Errors from the rest of the validation. Fixing this would require refactoring ValidateProject(), which is probably long overdue, but a lot of things touch that, so the scope of this somewhat urgent change grows dramatically if we go down that road.